### PR TITLE
Remove specialist topics from email-alert-service

### PR DESCRIPTION
**Pull request by unoduetre**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

Remove specialist topics from the supported attributes in  email-alert-service.

Please investigate and make a note on the PR about what the \`:topic\`  field is referring to here: [https://github.com/alphagov/email-alert-service/blob/53ce12e4ea8f5643dbb3051c8ec5c9651afa8b6b/lib/tasks/message_queues.rake#L10C45-L10C51](https://github.com/alphagov/email-alert-service/blob/53ce12e4ea8f5643dbb3051c8ec5c9651afa8b6b/lib/tasks/message_queues.rake#L10C45-L10C51 "smartCard-inline")

## Why

Retiring specialist topics.

[Trello card](https://trello.com/c/hIUgTP0C/2497-remove-specialist-topics-from-email-alert-service-s-m), [Jira issue NAV-12379](https://gov-uk.atlassian.net/browse/NAV-12379)

### CODE REVIEW

**Comment by unoduetre** [comment]
The :topic field mentioned in the requirement is unrelated and it’s a message type of RabbitMQ: https://www.rabbitmq.com/tutorials/amqp-concepts#exchange-topic

**Comment by unoduetre** [comment]
We've got some questions related to this PR, so those need to be answered first before it's merged, so I've marked it as "do not merge" until those are answered.

**Review by Tetrino — APPROVED** [review]

**Review by hannako — CHANGES_REQUESTED** [review]
I've made a couple of notes here. But it's the same comment as on the email-alert-api PR (so I'll spare you from having to read it again!) - ie let's use valid tag types, not legacy ones.

File: spec/models/email_alert_spec.rb
Comment (unoduetre) [reply spec/models/email_alert_spec.rb:17]: Removed

File: spec/models/email_alert_spec.rb
Comment (unoduetre) [reply spec/models/email_alert_spec.rb:113]: Removed

File: spec/models/email_alert_spec.rb
Comment (unoduetre) [reply spec/models/email_alert_spec.rb:128]: I've replaced policies **in links** with other things. It doesn't seem to be possible to remove policies from tags. If my understanding is correct, email-alert-service is used by publishing api. The list of supported links/tags is [here](https://github.com/alphagov/email-alert-service/blob/c2bbc08a16af48cd4b5f4fd8dcc57ca533fdbee5/email_alert_service/models/major_change_message_processor.rb#L67). The only **tag** in this list which is also used by publishing api is `policies`.

**Review by hannako — APPROVED** [review]
Looks good, thanks Mat

File: spec/models/email_alert_spec.rb
Comment (unoduetre) [reply spec/models/email_alert_spec.rb:128]: I think that, removing the tests for tags should go together with removing support for tags. Unless it's done this way, we will simply end up with untested, unused code.

Having said that, I don't think this is in the scope of the ticket behind this PR, which is about removing specialist topics. Removing tags seems like a separate, larger task and it would be better done by someone from the team responsible for this app who normally works on this code and understands it well (if there is such a person), instead of a person from another team. If there is no such person, I think it still should be another ticket. It is not really related to removing specialist topics.

With respect to using other kinds of tags, in another PR you explained that it would be good to make tests to reflect real data. Here the proposition is to go in the opposite direction - to introduce in the tests elements which cannot at this point reflect real data, if I understand things correctly. In some sense a deprecated field which was used in the past to me looks more like "real data" than a field which has never been used and can only be used potentially in the future.

From yet another perspective though, if there are confirmed plans to e.g. move the specialist publisher to publishing-api, I suppose it would make sense to use its data in tests as the preparation for the future.

What do you think?

Comment (KludgeKML) [reply spec/models/email_alert_spec.rb:128]: Eh, it's not really the opposite of the email-alert-api change - that was "can we make the data look more realistic for the tag", whereas this is "can we not use tags the don't exist". I mean, my ideal proposal here is more the last line: can we just get rid of the tests?

We're describing three options:
1) use tags that do exist (even if not currently valid)
2) get rid of the tags code altogether (and put it back if necessary later)
3) use tags that don't exist, 

If you don't want to do option 2, I think option 1 is clearly a better idea than 3

Comment (unoduetre) [reply spec/models/email_alert_spec.rb:128]: OK, I will use the specialist publisher tags as a replacement for policy.

Comment (unoduetre) [reply spec/models/email_alert_spec.rb:128]: @KludgeKML I've applied the changes.

**Review by KludgeKML — APPROVED** [review]
LGTM

**Pull request merged**
